lang/c++/build.sh

```diff
@@ -76,7 +76,24 @@ function do_dist() {
 
 case "$target" in
   lint)
-    echo 'This is a stub where someone can provide linting.'
+    # Variable that will hold the name of the clang-format command
+    CLANG_FORMAT=""
+
+    # Some distros just call it clang-format. Others (e.g. Ubuntu) are insistent
+    # that the version number be part of the command. We prefer clang-format if
+    # that's present, otherwise we work backwards from highest version to lowest
+    # version.
+    for clangfmt in clang-format{,-{4,3}.{9,8,7,6,5,4,3,2,1,0}}; do
```

Comment:
I think it would read better if you moved this search-for-command into its own function, something like

```sh
clangfmt() {
    local clangfmt
    for clangfmt in clang-format{,-{4,3}.{9,8,7,6,5,4,3,2,1,0}}; do
        if command -v "$clangfmt" &> /dev/null; then
            command "$clangfmt" "$@"
            return
        fi
    done
    echo 'Failed to find clang-format. Skipping lint.' >&2
    return 254
}
```

Then inside the lint command you can just call the `clangfmt` function directly.